cerberus-web/src/test/java/com/nike/cerberus/event/filter/AuditLoggingFilterTest.java

```diff
@@ -0,0 +1,109 @@
+package com.nike.cerberus.event.filter;
+
+import static org.junit.Assert.assertTrue;
+import static org.mockito.Mockito.*;
+import static org.mockito.MockitoAnnotations.initMocks;
+
+import com.nike.cerberus.util.SdbAccessRequest;
+import java.io.IOException;
+import javax.servlet.FilterChain;
+import javax.servlet.ServletException;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Test;
+import org.mockito.Mock;
+import org.mockito.Mockito;
+import org.springframework.boot.info.BuildProperties;
+import org.springframework.context.ApplicationEvent;
+import org.springframework.context.ApplicationEventPublisher;
+import org.springframework.security.core.Authentication;
+import org.springframework.security.core.context.SecurityContextHolder;
+
+public class AuditLoggingFilterTest {
+
+  AuditLoggingFilter auditLoggingFilter;
+  HttpServletRequest request;
+  HttpServletResponse response;
+  FilterChain filterChain;
+
+  @Mock private Authentication authentication;
+  private SdbAccessRequest sdbAccessRequest;
+  private AuditLoggingFilterDetails auditLoggingFilterDetails;
+  private BuildProperties buildProperties;
+  private ApplicationEventPublisher applicationEventPublisher;
+
+  @Before
+  public void setUp() {
+    initMocks(this);
+    request = mock(HttpServletRequest.class);
+    response = mock(HttpServletResponse.class);
+    filterChain = mock(FilterChain.class);
+    Mockito.when(authentication.getPrincipal()).thenReturn("principal");
+    SecurityContextHolder.getContext().setAuthentication(authentication);
+    sdbAccessRequest = mock(SdbAccessRequest.class);
+    auditLoggingFilterDetails = mock(AuditLoggingFilterDetails.class);
+    buildProperties = mock(BuildProperties.class);
+    applicationEventPublisher = mock(ApplicationEventPublisher.class);
+    auditLoggingFilter =
+        new AuditLoggingFilter(
+            sdbAccessRequest,
+            auditLoggingFilterDetails,
+            buildProperties,
+            applicationEventPublisher);
+  }
+
+  @Test
+  public void testShouldNotFilter() {
+    Mockito.when(request.getServletPath()).thenReturn("/dashboard/**");
+    assertTrue(auditLoggingFilter.shouldNotFilter(request));
+  }
+
+  @Test
+  public void testDoFilterInternal() throws IOException, ServletException {
+    doNothing().when(filterChain).doFilter(request, response);
+    when(request.getMethod()).thenReturn("GET");
+    when(request.getServletPath()).thenReturn("path");
+    mockxFwdAndVersion();
+    when(auditLoggingFilterDetails.getSdbNameSlug()).thenReturn("sdbNameSlug");
+    doNothing().when(applicationEventPublisher).publishEvent(ApplicationEvent.class);
+    auditLoggingFilter.doFilterInternal(request, response, filterChain);
```

Comment:
Shouldn't there be some kind of an assertion in this test?

Reply:
yeah we can add. actually i missed for void type methods. I will add it now.